google/src/main/scala/org/broadinstitute/dsde/workbench/util/DistributedLock.scala

```
@@ -38,14 +39,14 @@ class DistributedLock[F[_]](lockPathPrefix: String, config: DistributedLockConfi
         lockStatus <- getLockStatus(tx, docRef)
         _ <- lockStatus match {
           case Available => setLock(tx, docRef, lock.expiresIn)
-          case Locked => af.raiseError[Unit](new WorkbenchException(s"can't get lock: $lock"))
+          case Locked => asyncF.raiseError[Unit](new WorkbenchException(s"can't get lock: $lock"))
```

So it fails fast if the lock can't be acquired, as opposed to blocking (up to some timeout) waiting for the lock to be available?

---

this function is just for acquiring a lock. `withLock` will actually retry and semantically block